chore: update Supabase examples.

thorwebdev opened the pull request:
<!--
Thanks for opening a PR! Your contribution is much appreciated.
To make sure your PR is handled as smoothly as possible we request that you follow the checklist sections below.
Choose the right checklist for the change that you're making:
-->

Removing the outdated example and replacing it with a list of useful Next.js + Supabase examples and resources. Does that work for y'all?

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [x] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

leerob commented:
Unfortunately, this would create folks wanting to do `create-next-app -e with-supabase` type cloning, for example. I would recommend instead keeping an example up to date here 🙏 In the future, we will be removing some of the empty directory README only "examples".

thorwebdev commented:
That's a good point, thanks @leerob 

I've removed the old `with-supabase-auth-realtime-db` example and replaced it with an up to date `with-supabase` example. PTAL

Let me know if you're okay with the name change or if you'd prefer to stick with `with-supabase-auth-realtime-db`?  E.g. would it affect this listing: https://vercel.com/templates/next.js/supabase-authentication ?

balazsorban44 commented:
Closing in favor of #51335
